scvi/dataset/dataset.py

```diff
@@ -577,13 +577,14 @@ def initialize_cell_attribute(self, attribute_name, attribute, categorical=False
                 )
             )
             attribute_name = valid_attribute_name
-        if not self.nb_cells == len(attribute):
+        len_attribute = attribute.shape[0] if type(attribute) is not list else len(attribute)
+        if not self.nb_cells == len_attribute:
             raise ValueError(
                 "Number of cells ({n_cells}) and length of cell attribute ({n_attr}) mismatch".format(
-                    n_cells=self.nb_cells, n_attr=len(attribute)
+                    n_cells=self.nb_cells, n_attr=len_attribute
                 )
             )
-        setattr(self, attribute_name, np.asarray(attribute))
+        setattr(self, attribute_name, np.asarray(attribute) if not isinstance(attribute, csr_matrix) else attribute)
```

Comment:
good catch.
In general what's the number of measured protein markers?
I get the feeling that that number is pretty small, so we probably want to densify them in the dataset classes.

Reply:
it's about 100, no need to be sparse, but this will be necessary in the future.

Reply:
my point is, we should come up with a criterion to densify protein measurements in the dataset classes, like in the anndata loader

Reply:
Can you please add this kind of behaviour real quick? and fix the test, you have a missing `sp_sparse`. in front of `csr_matrix` somewhere apparently